File: 백준/Silver/10845. 큐/큐.c
#include <stdio.h>
#include <string.h>

int main(void) { 
   int a;
   scanf("%d", &a);

   int k, front = 0, back = 0, num[10000];
   
   for(int i = 0; i < a; i++){
      char arr[6];
      scanf("%s", arr);
      if(strcmp(arr, "push") == 0){
         scanf("%d", &k);
         num[back] = k;
         back += 1;
      }
      else{
         if(strcmp(arr, "front") == 0){
            if(back == front){
                printf("-1\n");
            }
            else {
                printf("%d\n", num[front]);
            }
         }
         else if(strcmp(arr, "back") == 0){
            if(back == front){
               printf("-1\n");
            }
            else {
                  printf("%d\n", num[back - 1]);
            }
         }
         else if(strcmp(arr, "pop") == 0){
            if(back == front){
               printf("-1\n");
            }
            else{
               printf("%d\n", num[front]);
               front += 1;
            }
         }
         else if(strcmp(arr, "size") == 0){
            printf("%d\n", back - front);
         }
         else{
            if(back == front){
               printf("1\n");
            }
            else{
               printf("0\n");
            }
         }
      }
   }
   
   return 0;
}

File: 백준/Silver/10845. 큐/큐.py
import sys

a = int(sys.stdin.readline())

arr = []

for _ in range(a):
  ip = sys.stdin.readline()
  ip = ip[0:-1]
  if ip[0:4] == "push":
    arr.append(ip[5:])

  elif ip[0:5] == "empty":
    if len(arr) == 0:
      print(1)
    else:
      print(0)

  elif ip[0:3] == "pop":
    if len(arr) == 0:
      print(-1)
    else:
      print(arr.pop(0))

  elif ip[0:5] == "front":
    if len(arr) == 0:
      print(-1)
    else:
      print(arr[0])

  elif ip[0:4] == "back":
    if len(arr) == 0:
      print(-1)
    else:
      print(arr[-1])

  else:
    print(len(arr))